test/unit/CitationFormat.test.js

```diff
@@ -0,0 +1,180 @@
+      work.editions[0].sub_title = 'edition subtitle';
+      const citationData = CitationFormatter.getCitationData(work, work.editions[0]);
+      expect(citationData.sub_title).to.equal('edition subtitle');
+    });
+
+    it('should set arrays of agents types', () => {
+      const citationData = CitationFormatter.getCitationData(work, work.editions[0]);
+      expect(citationData.agents.authors.length).to.equal(2);
+      expect(citationData.agents.authors).to.deep.equal(['agent1', 'agent2']);
+      expect(citationData.agents.editors.length).to.equal(2);
+      expect(citationData.agents.editors).to.deep.equal(['agent1', 'agent2']);
+      expect(citationData.agents.illustrators.length).to.equal(2);
+      expect(citationData.agents.illustrators).to.deep.equal(['agent1', 'agent2']);
+      expect(citationData.agents.translators.length).to.equal(2);
+      expect(citationData.agents.translators).to.deep.equal(['agent1', 'agent2']);
+      expect(citationData.agents.publishers.length).to.equal(2);
+      expect(citationData.agents.publishers).to.deep.equal(['agent1', 'agent2']);
+      expect(citationData.agents.contributors.length).to.equal(2);
+      expect(citationData.agents.contributors).to.deep.equal(['agent1', 'agent2']);
+    });
+
+    it('should set publication year from edition', () => {
+      const citationData = CitationFormatter.getCitationData(work, work.editions[0]);
+      expect(citationData.publication_year).to.equal('1852');
+    });
+
+    it('should set publication place from edition', () => {
+      const citationData = CitationFormatter.getCitationData(work, work.editions[0]);
+      expect(citationData.publication_place).to.equal('London');
+    });
+
+    it('should set edition_statement from edition', () => {
+      const citationData = CitationFormatter.getCitationData(work, work.editions[0]);
+      expect(citationData.edition).to.equal('[Another ed.].');
+    });
+
+    it('should set volume from edition', () => {
+      work.editions[0].volume = 'Test Volume';
+      const citationData = CitationFormatter.getCitationData(work, work.editions[0]);
+      expect(citationData.volume).to.equal('Test Volume');
+    });
+
+    it('should set series from work', () => {
+      const citationData = CitationFormatter.getCitationData(work, work.editions[0]);
+      expect(citationData.series).to.equal('The modern library classics');
+    });
+
+    it('should set a source link from the edition', () => {
+      const citationData = CitationFormatter.getCitationData(work, work.editions[0]);
+      expect(citationData.sourceLink).to.equal('test_link');
+    });
+
+    it('should set government document status', () => {
+      const citationData = CitationFormatter.getCitationData(work, work.editions[0]);
+      expect(citationData.isGovernmentDoc).to.equal(false);
+    });
+  });
+
+  describe('getAgentsOfType()', () => {
+    const testAgents = [
+      { name: 'test1', roles: ['test', 'other_again'] },
```

Comment:
can we have these values be some kind of representation of what the data actually might be, to help it serve better as documentation?